src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -101,6 +99,7 @@ public virtual void Start()
         {
             Initialize();
 
+            _applicationLifetime = _applicationServices.GetRequiredService<IApplicationLifetime>() as ApplicationLifetime;
```

Comment:
Is it better to throw here if IApplicationLifetime is not ApplicationLifetime. Hosting would not be able to use it anyway.